Working with roles and environments

Durf1985 opened the pull request:
# Сreate environments and roles

## In the process done:

* Created roles `app` and `db` with using `ansible-galaxy init`
* Transfer of old code in accordance with the structure of roles
* Created stage and prod environments
* Added dynamic inventory to each environment
* Installing a community role `jdauphant.nginx`
* Configuring `nginx` to work with `reddit monolith` as a reverse proxy
* Encryption of credentials with Ansible Vault
## How to launch a project:

Run command `ansible-playbook ~/your_repo/ansible/playbook/site.yml`


## How to check the performance:

In the browser, type in the address bar `http://app_vm_external_address:80`

## PR checklist

The label with the homework number, in label description - homework lecture and topic

Durf1985 committed:
DO-82 Create app/db roles

Instead of app.yml/db.yml playbooks, created roles that using split
files into functional chunks

Durf1985 committed:
DO-83 Create stage and prod environments

The directory structure has been changed. Files were sorted according
to the directory structure. Created 2 environments stage and prod

Durf1985 committed:
DO-84/85 Refactoring/using community nginx role

Added in ansible configuration jdauphant.nginx role

loktionovam commented on ansible/roles/app/tasks/main.yml:
```diff
@@ -0,0 +1,20 @@
+
+- name: Add config for DB connection
+  ansible.builtin.template:
+    src: templates/db_config.j2
```
Here you can use `db_config.j2` without `template` directory prefix (like you do when template `src: mongod.conf.j2` )

loktionovam commented on ansible/environments/prod/group_vars/app:
```diff
@@ -1 +1,7 @@
 db_host: "{{hostvars['reddit-db']['gce_private_ip']}}"
+nginx_sites:
+default:
```
Here is a problem with indentation

Durf1985 committed:
DO-86 Using ansible vault

Created credentials files
Encrypted credentials files
Addede new playbook users.yml

loktionovam commented on ansible/ansible.cfg:
```diff
@@ -9,6 +9,7 @@ remote_user = appuser
 private_key_file = ~/.ssh/appuser
 host_key_checking = False
 retry_files_enabled = False
+vault_password_file = ./vault.key
```
It is better to store vault key completely outside of your repo, for example in `~/.ansible/vault.key`. This prevents accidentally commit of your key.

Durf1985 replied:
fix
